src/applications/personalization/preferences/containers/SetPreferences.jsx

```diff
@@ -31,10 +31,7 @@ class SetPreferences extends React.Component {
 
   // if the preferences are saved successfully, then redirect to home
   componentDidUpdate(prevProps) {
-    if (
-      prevProps.preferences.saveStatus === LOADING_STATES.pending &&
-      this.props.preferences.saveStatus === LOADING_STATES.loaded
-    ) {
+    if (didJustSave(prevProps, this.props)) {
```

Comment:
May not work but it might be good to put this on the state directly e.g. `shouldRouteHome`. The component could check for that prop and dispatch an action that updates the state so that it reflects a  "should not route home" state.

@jbalboni something to consider for long term: we may want to handle redirects in middleware instead of components. not sure what the right pattern is here- 
- components check for redirect prop
- reducers add a redirect flag (not sure you can add middleware _after_ reducers)
- action creators add a redirect trigger 

My assumption is if you need to redirect, there's no point in involving a react component since it's handling display concerns- usually don't display something before redirecting. 

Downside is that handling redirects in middleware adds some indirection.